SubjectsDAL/Repos/BaseRepo.cs

```diff
@@ -43,8 +43,32 @@ internal int SaveChanges()
             }
         }
 
-        public T GetOne(int id) => table.Find(id);
+        public T GetOne(int? id) => table.Find(id);
 
+        public List<T> GetAll() => table.ToList();
 
+        public int Add(T entity)
+        {
+            table.Add(entity);
+            return SaveChanges();
+        }
+
+        public int AddRange(List<T> entities)
+        {
+            table.AddRange(entities);
+            return SaveChanges();
+        }
+
+        public int Delete(T entity)
+        {
+            context.Entry(entity).State = EntityState.Deleted;
```

Comment:
Why are you changing the state of the entity on your own instead of using a dedicated `EntityFramework` method for this purpose?